src/platform/intel/ace/platform.c

```diff
@@ -71,6 +75,57 @@ int platform_boot_complete(uint32_t boot_message)
 	return ipc_platform_send_msg(&msg);
 }
 
+
+/* address where zephyr PM will save memory during D3 transition */
+extern void *global_imr_ram_storage;
+
+/* notifier called before every power state transition */
+static void notify_pm_state_entry(enum pm_state state)
+{
+	size_t storage_buffer_size;
+
+	if (state == PM_STATE_SOFT_OFF) {
+		/* allocate IMR global_imr_ram_storage */
+		const struct device *tlb_dev = DEVICE_DT_GET(DT_NODELABEL(tlb));
+
+		__ASSERT_NO_MSG(tlb_dev != NULL);
```

Comment:
do we really want to assert() if PM state transitions fail ? I know this means we cant do the PM transition, but how would host know - we would just see a "IPC timeout" bug being registered here.

Reply:
@marcinszkudlinski ?
Thats what we could also use "status codes" for right?